testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/OIDCProtocolMappersTest.java

```diff
@@ -1862,14 +1853,85 @@ public void testStaticScopeUsingDynamicScopeFormatWithDedicatedMappers() {
         assertEquals("value", idToken.getOtherClaims().get("from-dedicated-mapper"));
 
         AccessToken accessToken = oauth.verifyToken(response.getAccessToken());
-        assertTrue(response.getScope().contains(scopeRep.getName()));
+        assertTrue(response.getScope().contains(expectedScopeName));
         assertNotNull(accessToken.getOtherClaims());
         assertNotNull(accessToken.getOtherClaims().get("from-mapper"));
         // claim mapped by client scope mapper
         assertEquals("value", accessToken.getOtherClaims().get("from-mapper"));
         assertNotNull(idToken.getOtherClaims().get("from-dedicated-mapper"));
         // claim mapped by dedicated client mapper
-        assertEquals("value", idToken.getOtherClaims().get("from-dedicated-mapper"));
+        assertEquals("value", accessToken.getOtherClaims().get("from-dedicated-mapper"));
+    }
+
+    @Test
+    public void testStaticScopeUsingDynamicScopeFormatPrefixedWithScopeAsDefaultScope() {
+        RealmResource realm = adminClient.realm("test");
+        ClientResource clientResource = findClientResourceByClientId(realm, "test-app");
+        ClientRepresentation client = clientResource.toRepresentation();
+
+        // make sure the name of the client maps to the prefix of the dynamic scope name
+        client.setName("test");
+        clientResource.update(client);
+
+        // creates a client scope using the dynamic scope format and add it to the client as optional scope
+        createClientScope(realm, clientResource, "test", "from-scope-mapper", "value", true);
+        createClientScope(realm, clientResource, "test:create", "from-dynamic-scope", "value", true);
+
+        oauth.scope("openid test:create");
+        AccessTokenResponse response = oauth.doPasswordGrantRequest("test-user@localhost", "password");
+        assertTrue(response.getScope().contains("test:create"));
+
+        AccessToken accessToken = oauth.verifyToken(response.getAccessToken());
+        assertTrue(response.getScope().contains("test:create"));
```

Comment:
This assertion appears to be duplicated from line 1882. The method below has the same duplication.